### ISSUE TITLE
Strange behaviour of mask_loss_iou of MS R-CNN

### ISSUE THREAD

**Issue by AKMourato** [open-mmlab/mmdetection#5880]
Greetings.
I'm getting a strange behaviour on the IoU loss of Mask Head of a pre-trained Mask Scoring R-CNN.
```python
2021-08-13 13:04:05,943 - mmdet - INFO - Epoch [1][50/1813]	lr: 5.934e-05, eta: 0:35:03, time: 1.193, data_time: 0.820, memory: 2898, loss_rpn_cls: 0.4130, loss_rpn_bbox: 0.0487, loss_cls: 0.3978, acc: 84.4570, loss_bbox: 0.2301, loss_mask: 0.6885, loss_mask_iou: 0.0001, loss: 1.7782
2021-08-13 13:04:59,494 - mmdet - INFO - Epoch [1][100/1813]	lr: 1.193e-04, eta: 0:32:19, time: 1.071, data_time: 0.695, memory: 3093, loss_rpn_cls: 0.1312, loss_rpn_bbox: 0.0440, loss_cls: 0.2096, acc: 92.4336, loss_bbox: 0.2655, loss_mask: 0.3451, loss_mask_iou: 0.0000, loss: 0.9954
2021-08-13 13:05:52,964 - mmdet - INFO - Epoch [1][150/1813]	lr: 1.792e-04, eta: 0:30:47, time: 1.069, data_time: 0.695, memory: 3219, loss_rpn_cls: 0.0974, loss_rpn_bbox: 0.0418, loss_cls: 0.1893, acc: 92.4551, loss_bbox: 0.2787, loss_mask: 0.1487, loss_mask_iou: 0.0000, loss: 0.7558
2021-08-13 13:06:46,873 - mmdet - INFO - Epoch [1][200/1813]	lr: 2.392e-04, eta: 0:29:38, time: 1.078, data_time: 0.663, memory: 3219, loss_rpn_cls: 0.0774, loss_rpn_bbox: 0.0399, loss_cls: 0.1884, acc: 92.8438, loss_bbox: 0.2956, loss_mask: 0.1244, loss_mask_iou: 0.0000, loss: 0.7258
2021-08-13 13:07:39,583 - mmdet - INFO - Epoch [1][250/1813]	lr: 2.991e-04, eta: 0:28:28, time: 1.054, data_time: 0.642, memory: 3302, loss_rpn_cls: 0.0775, loss_rpn_bbox: 0.0412, loss_cls: 0.1904, acc: 92.7520, loss_bbox: 0.3117, loss_mask: 0.1191, loss_mask_iou: 0.0001, loss: 0.7399
2021-08-13 13:08:33,731 - mmdet - INFO - Epoch [1][300/1813]	lr: 3.590e-04, eta: 0:27:31, time: 1.083, data_time: 0.689, memory: 3302, loss_rpn_cls: 0.0843, loss_rpn_bbox: 0.0423, loss_cls: 0.1877, acc: 93.0488, loss_bbox: 0.2815, loss_mask: 0.1047, loss_mask_iou: 0.0011, loss: 0.7016
2021-08-13 13:09:27,171 - mmdet - INFO - Epoch [1][350/1813]	lr: 4.190e-04, eta: 0:26:32, time: 1.069, data_time: 0.646, memory: 3302, loss_rpn_cls: 0.0717, loss_rpn_bbox: 0.0380, loss_cls: 0.1922, acc: 92.8086, loss_bbox: 0.3115, loss_mask: 0.0991, loss_mask_iou: 0.0026, loss: 0.7151
2021-08-13 13:10:18,560 - mmdet - INFO - Epoch [1][400/1813]	lr: 4.789e-04, eta: 0:25:26, time: 1.028, data_time: 0.598, memory: 3302, loss_rpn_cls: 0.0547, loss_rpn_bbox: 0.0339, loss_cls: 0.1697, acc: 93.8184, loss_bbox: 0.2775, loss_mask: 0.1028, loss_mask_iou: 0.0131, loss: 0.6518
2021-08-13 13:11:11,374 - mmdet - INFO - Epoch [1][450/1813]	lr: 5.389e-04, eta: 0:24:29, time: 1.056, data_time: 0.619, memory: 3320, loss_rpn_cls: 0.0648, loss_rpn_bbox: 0.0369, loss_cls: 0.1751, acc: 93.4766, loss_bbox: 0.2801, loss_mask: 0.0864, loss_mask_iou: 0.0187, loss: 0.6620
2021-08-13 13:12:04,417 - mmdet - INFO - Epoch [1][500/1813]	lr: 5.988e-04, eta: 0:23:33, time: 1.061, data_time: 0.654, memory: 3320, loss_rpn_cls: 0.0694, loss_rpn_bbox: 0.0403, loss_cls: 0.1780, acc: 93.4199, loss_bbox: 0.2805, loss_mask: 0.0805, loss_mask_iou: 0.0242, loss: 0.6728
2021-08-13 13:12:55,458 - mmdet - INFO - Epoch [1][550/1813]	lr: 6.000e-04, eta: 0:22:32, time: 1.021, data_time: 0.598, memory: 3320, loss_rpn_cls: 0.0673, loss_rpn_bbox: 0.0386, loss_cls: 0.1709, acc: 93.4414, loss_bbox: 0.2866, loss_mask: 0.0821, loss_mask_iou: 0.0232, loss: 0.6687
2021-08-13 13:13:48,288 - mmdet - INFO - Epoch [1][600/1813]	lr: 6.000e-04, eta: 0:21:37, time: 1.057, data_time: 0.600, memory: 3320, loss_rpn_cls: 0.0687, loss_rpn_bbox: 0.0387, loss_cls: 0.1805, acc: 93.0352, loss_bbox: 0.3254, loss_mask: 0.0791, loss_mask_iou: 0.0215, loss: 0.7140
2021-08-13 13:14:41,930 - mmdet - INFO - Epoch [1][650/1813]	lr: 6.000e-04, eta: 0:20:44, time: 1.073, data_time: 0.629, memory: 3320, loss_rpn_cls: 0.0673, loss_rpn_bbox: 0.0372, loss_cls: 0.1658, acc: 93.4668, loss_bbox: 0.3073, loss_mask: 0.0830, loss_mask_iou: 0.0179, loss: 0.6785
2021-08-13 13:15:32,811 - mmdet - INFO - Epoch [1][700/1813]	lr: 6.000e-04, eta: 0:19:46, time: 1.018, data_time: 0.562, memory: 3333, loss_rpn_cls: 0.0632, loss_rpn_bbox: 0.0355, loss_cls: 0.1810, acc: 93.1094, loss_bbox: 0.3073, loss_mask: 0.0803, loss_mask_iou: 0.0170, loss: 0.6843
2021-08-13 13:16:23,551 - mmdet - INFO - Epoch [1][750/1813]	lr: 6.000e-04, eta: 0:18:49, time: 1.015, data_time: 0.571, memory: 3400, loss_rpn_cls: 0.0481, loss_rpn_bbox: 0.0283, loss_cls: 0.1440, acc: 94.5664, loss_bbox: 0.2489, loss_mask: 0.0864, loss_mask_iou: 0.0164, loss: 0.5722
2021-08-13 13:17:11,571 - mmdet - INFO - Epoch [1][800/1813]	lr: 6.000e-04, eta: 0:17:50, time: 0.961, data_time: 0.555, memory: 3400, loss_rpn_cls: 0.0514, loss_rpn_bbox: 0.0329, loss_cls: 0.1461, acc: 94.6230, loss_bbox: 0.2376, loss_mask: 0.0859, loss_mask_iou: 0.0143, loss: 0.5683
2021-08-13 13:18:04,914 - mmdet - INFO - Epoch [1][850/1813]	lr: 6.000e-04, eta: 0:16:58, time: 1.067, data_time: 0.607, memory: 3400, loss_rpn_cls: 0.0623, loss_rpn_bbox: 0.0363, loss_cls: 0.1638, acc: 93.9355, loss_bbox: 0.2780, loss_mask: 0.0743, loss_mask_iou: 0.0145, loss: 0.6292
2021-08-13 13:19:01,539 - mmdet - INFO - Epoch [1][900/1813]	lr: 6.000e-04, eta: 0:16:09, time: 1.133, data_time: 0.673, memory: 3400, loss_rpn_cls: 0.0517, loss_rpn_bbox: 0.0362, loss_cls: 0.1758, acc: 93.0840, loss_bbox: 0.3170, loss_mask: 0.0665, loss_mask_iou: 0.0143, loss: 0.6614
2021-08-13 13:19:54,724 - mmdet - INFO - Epoch [1][950/1813]	lr: 6.000e-04, eta: 0:15:16, time: 1.063, data_time: 0.616, memory: 3400, loss_rpn_cls: 0.0597, loss_rpn_bbox: 0.0328, loss_cls: 0.1699, acc: 93.5195, loss_bbox: 0.2949, loss_mask: 0.0710, loss_mask_iou: 0.0139, loss: 0.6423
2021-08-13 13:20:48,806 - mmdet - INFO - Epoch [1][1000/1813]	lr: 6.000e-04, eta: 0:14:23, time: 1.082, data_time: 0.642, memory: 3400, loss_rpn_cls: 0.0519, loss_rpn_bbox: 0.0328, loss_cls: 0.1718, acc: 93.4043, loss_bbox: 0.3009, loss_mask: 0.0693, loss_mask_iou: 0.0146, loss: 0.6413
2021-08-13 13:21:43,207 - mmdet - INFO - Epoch [1][1050/1813]	lr: 6.000e-04, eta: 0:13:31, time: 1.088, data_time: 0.628, memory: 3400, loss_rpn_cls: 0.0575, loss_rpn_bbox: 0.0344, loss_cls: 0.1604, acc: 93.9043, loss_bbox: 0.2840, loss_mask: 0.0704, loss_mask_iou: 0.0120, loss: 0.6187
2021-08-13 13:22:37,118 - mmdet - INFO - Epoch [1][1100/1813]	lr: 6.000e-04, eta: 0:12:38, time: 1.078, data_time: 0.639, memory: 3400, loss_rpn_cls: 0.0542, loss_rpn_bbox: 0.0343, loss_cls: 0.1588, acc: 94.1934, loss_bbox: 0.2857, loss_mask: 0.0731, loss_mask_iou: 0.0112, loss: 0.6173
2021-08-13 13:23:28,679 - mmdet - INFO - Epoch [1][1150/1813]	lr: 6.000e-04, eta: 0:11:44, time: 1.031, data_time: 0.578, memory: 3400, loss_rpn_cls: 0.0547, loss_rpn_bbox: 0.0324, loss_cls: 0.1590, acc: 93.9688, loss_bbox: 0.2855, loss_mask: 0.0732, loss_mask_iou: 0.0135, loss: 0.6183
2021-08-13 13:24:28,513 - mmdet - INFO - Epoch [1][1200/1813]	lr: 6.000e-04, eta: 0:10:54, time: 1.197, data_time: 0.759, memory: 3400, loss_rpn_cls: 0.0469, loss_rpn_bbox: 0.0315, loss_cls: 0.1556, acc: 94.0195, loss_bbox: 0.2860, loss_mask: 0.0696, loss_mask_iou: 0.0117, loss: 0.6014
2021-08-13 13:25:17,763 - mmdet - INFO - Epoch [1][1250/1813]	lr: 6.000e-04, eta: 0:09:59, time: 0.985, data_time: 0.534, memory: 3400, loss_rpn_cls: 0.0450, loss_rpn_bbox: 0.0295, loss_cls: 0.1631, acc: 93.5684, loss_bbox: 0.2803, loss_mask: 0.0620, loss_mask_iou: 0.0135, loss: 0.5933
2021-08-13 13:26:12,371 - mmdet - INFO - Epoch [1][1300/1813]	lr: 6.000e-04, eta: 0:09:06, time: 1.092, data_time: 0.613, memory: 3400, loss_rpn_cls: 0.0567, loss_rpn_bbox: 0.0382, loss_cls: 0.1691, acc: 93.4316, loss_bbox: 0.3187, loss_mask: 0.0631, loss_mask_iou: 0.0124, loss: 0.6582

2021-08-13 13:27:05,242 - mmdet - INFO - Epoch [1][1350/1813]	lr: 6.000e-04, eta: 0:08:13, time: 1.058, data_time: 0.632, memory: 3414, loss_rpn_cls: 0.0552, loss_rpn_bbox: 0.0332, loss_cls: 0.1592, acc: 93.7656, loss_bbox: 0.2968, loss_mask: 0.0714, loss_mask_iou: 0.0121, loss: 0.6279
2021-08-13 13:28:00,109 - mmdet - INFO - Epoch [1][1400/1813]	lr: 6.000e-04, eta: 0:07:20, time: 1.097, data_time: 0.659, memory: 3414, loss_rpn_cls: 0.0630, loss_rpn_bbox: 0.0362, loss_cls: 0.1709, acc: 93.4844, loss_bbox: 0.3075, loss_mask: 0.0654, loss_mask_iou: 0.0129, loss: 0.6559
2021-08-13 13:28:53,293 - mmdet - INFO - Epoch [1][1450/1813]	lr: 6.000e-04, eta: 0:06:27, time: 1.064, data_time: 0.626, memory: 3414, loss_rpn_cls: 0.0579, loss_rpn_bbox: 0.0344, loss_cls: 0.1615, acc: 93.8789, loss_bbox: 0.2783, loss_mask: 0.0678, loss_mask_iou: 0.0120, loss: 0.6120
2021-08-13 13:29:45,393 - mmdet - INFO - Epoch [1][1500/1813]	lr: 6.000e-04, eta: 0:05:33, time: 1.042, data_time: 0.597, memory: 3414, loss_rpn_cls: 0.0493, loss_rpn_bbox: 0.0288, loss_cls: 0.1450, acc: 94.3809, loss_bbox: 0.2582, loss_mask: 0.0676, loss_mask_iou: 0.0121, loss: 0.5610
2021-08-13 13:30:42,766 - mmdet - INFO - Epoch [1][1550/1813]	lr: 6.000e-04, eta: 0:04:41, time: 1.148, data_time: 0.695, memory: 3414, loss_rpn_cls: 0.0713, loss_rpn_bbox: 0.0418, loss_cls: 0.1728, acc: 93.4023, loss_bbox: 0.3342, loss_mask: 0.0659, loss_mask_iou: 0.0117, loss: 0.6975
2021-08-13 13:31:38,037 - mmdet - INFO - Epoch [1][1600/1813]	lr: 6.000e-04, eta: 0:03:47, time: 1.105, data_time: 0.672, memory: 3414, loss_rpn_cls: 0.0582, loss_rpn_bbox: 0.0384, loss_cls: 0.1667, acc: 93.4531, loss_bbox: 0.3114, loss_mask: 0.0622, loss_mask_iou: 0.0127, loss: 0.6496
2021-08-13 13:32:33,909 - mmdet - INFO - Epoch [1][1650/1813]	lr: 6.000e-04, eta: 0:02:54, time: 1.118, data_time: 0.656, memory: 3414, loss_rpn_cls: 0.0524, loss_rpn_bbox: 0.0321, loss_cls: 0.1590, acc: 93.8340, loss_bbox: 0.2856, loss_mask: 0.0655, loss_mask_iou: 0.0125, loss: 0.6071
2021-08-13 13:33:30,741 - mmdet - INFO - Epoch [1][1700/1813]	lr: 6.000e-04, eta: 0:02:01, time: 1.136, data_time: 0.673, memory: 3414, loss_rpn_cls: 0.0570, loss_rpn_bbox: 0.0404, loss_cls: 0.1694, acc: 93.4766, loss_bbox: 0.3071, loss_mask: 0.0671, loss_mask_iou: 0.0138, loss: 0.6546
2021-08-13 13:34:21,297 - mmdet - INFO - Epoch [1][1750/1813]	lr: 6.000e-04, eta: 0:01:07, time: 1.011, data_time: 0.556, memory: 3414, loss_rpn_cls: 0.0430, loss_rpn_bbox: 0.0331, loss_cls: 0.1481, acc: 94.3535, loss_bbox: 0.2758, loss_mask: 0.0652, loss_mask_iou: 0.0119, loss: 0.5772
2021-08-13 13:35:19,109 - mmdet - INFO - Epoch [1][1800/1813]	lr: 6.000e-04, eta: 0:00:13, time: 1.156, data_time: 0.689, memory: 3414, loss_rpn_cls: 0.0674, loss_rpn_bbox: 0.0423, loss_cls: 0.1659, acc: 93.5586, loss_bbox: 0.3168, loss_mask: 0.0680, loss_mask_iou: 0.0117, loss: 0.6720
2021-08-13 13:35:33,863 - mmdet - INFO - Saving checkpoint at 1 epochs
```
The loss value is 0 until the 250th batch, and starts to increase from then on while the mask loss decreases.
Is there any reasonable explanation for this event?

**Comment by PeterVennerstrom** [comment]
From the paper:

```Training: For training the MaskIoU head, we use the RPN proposals as training samples. The training samples are required to have a IoU between proposal box and the matched ground truth box larger than 0.5```

It probably takes a few training steps before many proposal boxes with IoU greater than .5 are predicted so there are few or no training samples for MaskIoU loss at the beginning.
